soroban-sdk/src/crypto.rs

```diff
@@ -22,6 +22,28 @@ impl Crypto {
         unsafe { BytesN::unchecked_new(env.clone(), bin) }
     }
 
+    // Reseeds the pseudorandom number generator (PRNG) with the provided `seed` value.
+    pub fn prng_reseed(&self, seed: &Bytes) {
+        let env = self.env();
```

Comment:
I would add `env.check_same_env(seed.env())` to make sure these two are from the same `env`.

Reply:
I notice this check is not present in other functions but they are nice (and also inexpensive) to have .

Reply:
Cool, I added those checks [here](https://github.com/stellar/rs-soroban-sdk/pull/1023/commits/05e9e686fa33670189a4394669798f106caa4dea) if we want them. Easy to revert if not.

Reply:
I don't feel strongly about these existing or not; as @jayz22 says they are not meaningful in production code, only testing, and it's likely that if you have two hosts in a test you've got bigger problems. but at the same time it's harmless to add them.